sklearn/model_selection/_prediction.py

```diff
@@ -0,0 +1,462 @@
+from inspect import signature
+import numbers
+
+import numpy as np
+from joblib import Parallel, delayed
+
+from ._split import check_cv
+from ._split import StratifiedShuffleSplit
+
+from ..base import clone
+from ..base import BaseEstimator
+from ..base import ClassifierMixin
+from ..base import MetaEstimatorMixin
+from ..metrics import balanced_accuracy_score
+from ..metrics import roc_curve
+from ..preprocessing import LabelEncoder
+from ..utils import check_array
+from ..utils import _safe_indexing
+from ..utils.multiclass import check_classification_targets
+from ..utils.multiclass import type_of_target
+from ..utils.validation import check_is_fitted
+
+
+class CutoffClassifier(MetaEstimatorMixin, ClassifierMixin, BaseEstimator):
+    """Decision threshold calibration for binary classification.
+
+    Estimator that calibrates the decision threshold (cutoff point) that is
+    used for prediction. The methods for picking cutoff points make use of
+    traditional binary classification evaluation statistics such as the true
+    positive and true negative rates or any metrics accepting true labels and
+    the output of a scoring function from a scikit-learn estimator.
+
+    Parameters
+    ----------
+    base_estimator : estimator object
+        The classifier, fitted or not fitted, from which we want to optimize
+        the decision threshold used during `predict`.
+
+    objective_metric : callable or {"tpr", "tnr"} \
+            default=balanced_accuracy_score
+        The objective metric to be optimized. Can be one of:
+
+        * a callable with the signature `metric(y_true, y_score, **kwargs)`;
+        * `"tpr"`: find the decision threshold for a true positive ratio (TPR)
+          of `objective_value`;
+        * `"tnr"`: find the decision threshold for a true negative ratio (TNR)
+          of `objective_value`.
+
+    objective_metric_params : dict, default=None
+        Some extra parameters to pass to `objective_metric`.
+
+    objective_value : float, default=None
+        The value associated with the `objective_metric` metric for which we
+        want to find the decision threshold when `objective_metric` is equal to
+        `"tpr"` or `"tnr"`.
```

Comment:
What does None do if tpr or tnr is passed? Looking briefly I'm not sure it raises an error?